src/pages/index.js

```diff
@@ -123,13 +153,53 @@ function getcardElement(data) {
     previewModalImageEl.alt = data.name;
   });
 
-  cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
-  });
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  function handleLike(evt, cardId) {
+    const isLiked = evt.target.classList.contains("card__like-button_liked");
+    api
+      .handleLike(cardId, isLiked)
+      .then(() => {
+        cardLikeBtn.classList.toggle("card__like-button_liked");
+      })
+      .catch(console.error);
+  }
+
+  function handleDeleteCard(cardElement, cardId) {
+    openModal(deleteModal);
+    selectedCard = cardElement;
+    selectedCardId = cardId;
+  }
 
   return cardElement;
 }
 
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const cardDeleteBtn = evt.submitter;
+  const cardCancelBtn = deleteForm.querySelector(
+    ".modal__submit-button-cancel"
+  );
+
+  if (evt.submitter === cardDeleteBtn) {
+    setDeleteText(cardDeleteBtn, true);
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        setDeleteText(cardDeleteBtn, false);
+      });
+  } else {
+    closeModal(deleteModal);
+  }
+}
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
 }
```

Comment:
* According to the previous task, modals must get closed by pressing the `Escape` key and by clicking on the `overlay` https://snipboard.io/8fuqoI.jpg